lib/debug_logging/class_notifier.rb

```diff
@@ -1,53 +1,37 @@
+        base.extend(LambDartable::Note)
+      end
+    end
+
     def notified(*methods_to_notify)
       methods_to_notify, payload, config_opts = DebugLogging::Util.extract_payload_and_config(
         method_names: methods_to_notify,
         payload: nil,
         config: nil,
       )
-      Array(methods_to_notify).each do |method_to_notify|
-        method_to_notify, method_payload, method_config_opts = DebugLogging::Util.extract_payload_and_config(
-          method_names: method_to_notify,
+      Array(methods_to_notify).each do |decorated_method|
+        decorated_method, method_payload, method_config_opts = DebugLogging::Util.extract_payload_and_config(
+          method_names: decorated_method,
           payload: payload,
           config: config_opts,
         )
-        original_method = method(method_to_notify)
+        original_method = method(decorated_method)
         (class << self; self; end).class_eval do
-          define_method(method_to_notify) do |*args, &block|
-            config_proxy = DebugLogging::Util.config_proxy_finder(
-              scope: self,
-              config_opts: method_config_opts,
-              method_name: method_to_notify,
-              proxy_ref: "kn",
-            ) do |proxy|
-              ActiveSupport::Notifications.subscribe(
-                DebugLogging::ArgumentPrinter.debug_event_name_to_s(method_to_notify: method_to_notify),
-              ) do |*debug_args|
-                proxy.log do
-                  DebugLogging::LogSubscriber.log_event(ActiveSupport::Notifications::Event.new(*debug_args))
-                end
-              end
-            end
-            paydirt = DebugLogging::Util.payload_instance_variable_hydration(scope: self, payload: method_payload)
-            ActiveSupport::Notifications.instrument(
-              DebugLogging::ArgumentPrinter.debug_event_name_to_s(method_to_notify: method_to_notify),
-              {
-                debug_args: args,
-                config_proxy: config_proxy,
-                **paydirt,
-              },
-            ) do
-              if args.size == 1 && (harsh = args[0]) && harsh.is_a?(Hash)
-                original_method.call(**harsh, &block)
-              else
-                original_method.call(*args, &block)
-              end
-            rescue StandardError => e
-              if config_proxy.error_handler_proc
-                config_proxy.error_handler_proc.call(config_proxy, e, self, method_to_notify, args)
-              else
-                raise e
+          define_method(decorated_method) do |*args, **kwargs, &block|
+            lamb_dart = LambDart::Note.new(
+              klass: self,
+              method_config_opts:,
+              method_payload:,
+              args:,
+              kwargs:,
+              decorated_method:,
+            )
+            _dl_ld_notify(lamb_dart) do
+              _dl_ld_error_handle(lamb_dart) do
+                original_method.call(*args, **kwargs, &block)
```

Comment:
Perhaps we can do the same here as mentioned in the [above comment](https://github.com/pboling/debug_logging/pull/127/files#r1600709689)

Reply:
Agreed, ideally we could normalize them all, and DRY them up.